### PR TITLE
feat(ereal): standardize math/functions classify/numerics/truncate/minmax/next (#950 cat 1)

### PR DESCRIPTION
**Pull request by Ravenwater**
## Summary
First of ~5 category PRs decomposing the `elastic/ereal/math/functions/` refactor (#950), per the issue's "split by category" guidance. Aligns five classification / numeric-support files to the canonical test structure from `arithmetic/addition.cpp` (PR #943).

Relates to #944 (parent epic, Phase B). Relates to #950 (partial -- category 1 of 5).

## Changes (5 files)
- `classify.cpp` -- isfinite/isnan/isinf/isnormal/signbit/fpclassify
- `numerics.cpp` -- copysign/ldexp/frexp + frexp/ldexp roundtrip
- `truncate.cpp` -- floor/ceil/trunc/round/rint
- `minmax.cpp` -- min/max
- `next.cpp` -- nextafter

Each file:
- anonymous namespace + single `using namespace sw::universal;` (was nested `namespace sw { namespace universal { ... } }`)
- `reportTestCases`-gated `std::cout << "    FAIL ..."` (was unconditional-prefix `std::cerr << "FAIL: ..."`)
- `reportTestCases` defaults to `true` in main

No test logic changed; unqualified math calls resolve via ADL on the `ereal` argument. Each file stays single-family.

## Test Results
| Target | gcc build | gcc test | clang build | clang test |
|--------|-----------|----------|-------------|------------|
| er_math_classify | OK | PASS | OK | PASS |
| er_math_numerics | OK | PASS | OK | PASS |
| er_math_truncate | OK | PASS | OK | PASS |
| er_math_minmax | OK | PASS | OK | PASS |
| er_math_next | OK | PASS | OK | PASS |

## Remaining categories (separate PRs)
2. exponent/logarithm · 3. trigonometry/hyperbolic · 4. pow/sqrt/hypot · 5. fractional/error_and_gamma + remove stub. (Unicode stripping lands in categories 2-4.)

## Test plan
- [ ] Fast CI passes (gcc + clang CI_LITE)
- [ ] Promote to ready when satisfied: `gh pr ready <NNN>`

Relates to #950

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Test runners now enable per-case diagnostics by default, producing clearer conditional failure output during regressions.
  * Added property-based/fuzz tests across math function suites (classification, min/max, next/nextafter, numerics, truncate) to catch edge-case inconsistencies.
  * Improved test reporting and verification for numerical primitives to increase regression coverage.

[Review Change Stack](https://app.coderabbit.ai/change-stack/stillwater-sc/universal/pull/974?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: elastic/ereal/math/functions/next.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ namespace sw {
 	    result    = nextafter(x, y);
 	    if (result != expected) {
 		    if (reportTestCases) {
-			    std::cerr << "FAIL: nextafter(1.0, 2.0) != 1.0 + ulp(1.0)\n";
+			    std::cout << "    FAIL nextafter(1.0, 2.0) != 1.0 + ulp(1.0)\n";
 			    std::cerr << "  expected: " << to_binary(expected) << " : " << expected << '\n';
 			    std::cerr << "    result: " << to_binary(result) << " : " << result << '\n';
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply elastic/ereal/math/functions/next.cpp]: Fixed in 7935be46 -- the secondary diagnostic lines in next.cpp now use std::cout to match the FAIL lines. That commit also adds layer-1 property/special-value coverage (fuzzers) to all five files per a coverage-strategy decision.
